pkgs/servers/mattermost/default.nix

```diff
@@ -1,101 +1,51 @@
+  };
 
-    ldflags = [
-      "-s" "-w"
-      "-X ${goPackagePath}/model.BuildNumber=${version}${lib.optionalString (versionSuffix != null) "-${versionSuffix}"}"
-      "-X ${goPackagePath}/model.BuildDate=${buildDate}"
-      "-X ${goPackagePath}/model.BuildEnterpriseReady=false"
-    ];
+  webapp = fetchurl {
+    url = "https://releases.mattermost.com/${version}/mattermost-${version}-linux-amd64.tar.gz";
+    sha256 = "sha256-NWVDPDqdx7mdWCr/qBi8HsUsCJbT63x37UZGecHsZr4=";
   };
 
-  mattermost-server = if storePathAsBuildHash then mattermost-server-build.overrideAttrs (orig: {
-    buildPhase = ''
-      origGo="$(type -p go)"
+  vendorSha256 = "sha256-0sKuk0klxeep8J96RntDP9DHsVM4vrOmsKXiaWurVis=";
 
-      # Override the Go binary to set the build hash in -ldflags to $out.
-      # Technically this is more accurate than a Git hash!
-      # nixpkgs does not appear to support environment variables in ldflags
-      # for go packages, so we have to rewrite -ldflags before calling go.
-      go() {
-        local args=()
-        local ldflags="-X ${goPackagePath}/model.BuildHash=$out"
-        local found=0
-        for arg in "$@"; do
-          if [[ "$arg" == -ldflags=* ]] && [ $found -eq 0 ]; then
-            arg="-ldflags=''${ldflags} ''${arg#-ldflags=}"
-            found=1
-          fi
-          args+=("$arg")
-        done
-        "$origGo" "''${args[@]}"
-      }
+  subPackages = [ "cmd/mattermost" ];
 
-      ${orig.buildPhase}
-    '';
-  }) else mattermost-server-build;
+  ldflags = [
+    "-s"
+    "-w"
+    "-X github.com/mattermost/mattermost-server/v6/model.Version=${version}"
+  ];
 
-  mattermost-webapp = stdenv.mkDerivation {
-    pname = "mattermost-webapp";
-    inherit version;
+  postInstall = ''
+    tar --strip 1 --directory $out -xf $webapp \
+      mattermost/{client,i18n,fonts,templates,config}
 
-    src = fetchurl {
-      url = "https://releases.mattermost.com/${version}/mattermost-${version}-linux-amd64.tar.gz";
-      sha256 = "JDsCDZEtbeBTYuzOSwrxFNRKXy+9KXirjaCz6ODP5uw=";
-    };
+    # For some reason a bunch of these files are executable
+    find $out/{client,i18n,fonts,templates,config} -type f -exec chmod -x {} \;
+  '';
 
-    installPhase = ''
-      mkdir -p $out
-      tar --strip 1 --directory $out -xf $src \
-        mattermost/client \
-        mattermost/i18n \
-        mattermost/fonts \
-        mattermost/templates \
-        mattermost/config
+  passthru.tests.mattermost = nixosTests.mattermost;
 
-      # For some reason a bunch of these files are +x...
-      find $out -type f -exec chmod -x {} \;
-    '';
+  meta = with lib; {
+    description = "Mattermost is an open source platform for secure collaboration across the entire software development lifecycle";
+    homepage = "https://www.mattermost.org";
+    license = with licenses; [ agpl3 asl20 ];
+    maintainers = with maintainers; [ fpletz ryantm numinit kranzes ];
+    platforms = [ "x86_64-linux" ];
```

Comment:
Why is this restriction necessary? The go code should work on every platform we support and the webapp should be architecture-agnostic, right?

Reply:
You're right.